### ISSUE TITLE
PoolStringArray can not be inside of Array

### ISSUE THREAD

**Issue by Merlin1846** [godotengine/godot#57176]
### Godot version

3.4.2

### System information

UnExported Project Ubuntu 21.04

### Issue description

It does not appear to be possible to add values to a PoolStringArray that is inside of an Array. It does not show and error, stop the program or do anything else it just doesn't put it in the PoolStringArray.

### Steps to reproduce

```
func _ready():
    var test:Array = [PoolStringArray()]
    print(test)
    #Outputs an empty array within an array.
    test[0].append("Example String")
    print(test)
    print(test[0])
    #Outputs an empty array within an array. But should instead output [["Example String"]]
```

**Comment by akien-mga** [comment]
That's a known issue fixed in 4.0, and which is a design limitation in 3.x, see #15265.

You should store the `PoolStringArray` in a temporary variable to modify it:
```gdscript
var arr = test[0]
arr.append("Example String")
test[0] = arr
```

**Comment by Merlin1846** [comment]
Ok just making sure, I wanted to do it because the strings can get pretty long but I've already worked around it by having an Array in the Array.
